student.rb

```diff
@@ -1,15 +1,11 @@
-class Student
-  attr_reader :age, :phase
-  attr_accessor :name
+require_relative 'teach_learn'
 
-  def initialize(options = {})
-    @phase = 1
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
-  end
+class Student < TeachLearn
```

Comment:
A Student is a TeachLearn?  If you can't say a sentence like that and have it make sense, work on your naming.  A Student and a Teacher are both kinds of People (or DBCers or whatever), so I'd call the superclass `Person`.